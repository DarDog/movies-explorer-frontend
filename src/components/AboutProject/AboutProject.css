.about {
  padding: 110px 70px;
  background-color: #FFFFFF;
}

.about__list {
  gap: 40px;
}

.about__item {
  max-width: 550px;
}

.list__title {
  margin: 0 0 26px;
  padding: 0;
  font-weight: 400;
  font-size: 20px;
  line-height: 24px;
  letter-spacing: -0.04em;
  color: black;
}

.list__paragraph {
  margin: 0;
  padding: 0;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  color: black;
}

.about__track {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  margin-top: 110px;
}

.track__element {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.track__element_type_back-end {
  grid-column: 1;
}

.track__element_type_front-end {
  grid-column: 2/6;
}

.track__bar {
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 0 14px;
  padding: 0;
  width: 100%;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  color: black;
}

.track__bar_type_back-end {
  background-color: #2BE080;
}

.track__bar_type_front-end {
  background-color: #F2F2F2;
}

.track__bar-name {
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  color: #A0A0A0;
}

@media screen and (max-width: 768px){
  .about {
    padding: 90px 50px;
  }

  .about__list {
    gap: 30px;
  }

  .list__paragraph {
    font-size: 12px;
    line-height: 18px;
  }
}
