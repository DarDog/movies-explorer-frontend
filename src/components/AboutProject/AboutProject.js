import React from 'react';
import './AboutProject.css'
import SectionTitle from "../SectionTitle/SectionTitle";
import Ul from "../Ul/Ul";

const AboutProject = () => {
  return(
    <section className='about' id='about-project'>
      <SectionTitle>О проекте</SectionTitle>
      <Ul anotherClass='about__list'>
        <li className='about__item'>
          <h3 className='list__title'>Дипломный проект включает 5 этапов</h3>
          <p className='list__paragraph'>Составление плана, работу над бэкендом, вёрстку, добавление функциональности
            и финальные доработки.</p>
        </li>
        <li className='about__item'>
          <h3 className='list__title'>На выполнение диплома ушло 5 недель</h3>
          <p className='list__paragraph'>У каждого этапа был мягкий и жёсткий дедлайн, которые нужно было соблюдать,
            чтобы успешно защититься.</p>
        </li>
      </Ul>
      <div className='about__track'>
        <div className='track__element track__element_type_back-end'>
          <div className='track__bar track__bar_type_back-end'>1 неделя</div>
          <p className='track__bar-name'>Back-end</p>
        </div>
        <div className='track__element track__element_type_front-end'>
          <div className='track__bar track__bar_type_front-end'>4 недели</div>
          <p className='track__bar-name'>Front-end</p>
        </div>
      </div>
    </section>
  );
}

export default AboutProject;
