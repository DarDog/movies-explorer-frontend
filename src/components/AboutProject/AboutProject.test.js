import React from 'react';
import AboutProject from "./AboutProject";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';

describe('Компонент AboutProject', () => {
  it('Рендерится', () => {
    render(<AboutProject />);

    expect(screen.getByRole('heading', {level: 2})).toBeInTheDocument();
    expect(screen.getByText('О проекте')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('heading', {level: 3})).toHaveLength(2);
  })
  it('Соответствует своему snapshot', () => {
    const component = render(<AboutProject />);

    expect(component).toMatchSnapshot();
  })
})
