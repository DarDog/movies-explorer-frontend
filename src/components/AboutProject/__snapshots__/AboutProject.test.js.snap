// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент AboutProject Соответствует своему snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="about"
        id="about-project"
      >
        <h2
          class="title"
        >
          О проекте
        </h2>
        <ul
          class="list about__list "
        >
          <li
            class="about__item"
          >
            <h3
              class="list__title"
            >
              Дипломный проект включает 5 этапов
            </h3>
            <p
              class="list__paragraph"
            >
              Составление плана, работу над бэкендом, вёрстку, добавление функциональности и финальные доработки.
            </p>
          </li>
          <li
            class="about__item"
          >
            <h3
              class="list__title"
            >
              На выполнение диплома ушло 5 недель
            </h3>
            <p
              class="list__paragraph"
            >
              У каждого этапа был мягкий и жёсткий дедлайн, которые нужно было соблюдать, чтобы успешно защититься.
            </p>
          </li>
        </ul>
        <div
          class="about__track"
        >
          <div
            class="track__element track__element_type_back-end"
          >
            <div
              class="track__bar track__bar_type_back-end"
            >
              1 неделя
            </div>
            <p
              class="track__bar-name"
            >
              Back-end
            </p>
          </div>
          <div
            class="track__element track__element_type_front-end"
          >
            <div
              class="track__bar track__bar_type_front-end"
            >
              4 недели
            </div>
            <p
              class="track__bar-name"
            >
              Front-end
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="about"
      id="about-project"
    >
      <h2
        class="title"
      >
        О проекте
      </h2>
      <ul
        class="list about__list "
      >
        <li
          class="about__item"
        >
          <h3
            class="list__title"
          >
            Дипломный проект включает 5 этапов
          </h3>
          <p
            class="list__paragraph"
          >
            Составление плана, работу над бэкендом, вёрстку, добавление функциональности и финальные доработки.
          </p>
        </li>
        <li
          class="about__item"
        >
          <h3
            class="list__title"
          >
            На выполнение диплома ушло 5 недель
          </h3>
          <p
            class="list__paragraph"
          >
            У каждого этапа был мягкий и жёсткий дедлайн, которые нужно было соблюдать, чтобы успешно защититься.
          </p>
        </li>
      </ul>
      <div
        class="about__track"
      >
        <div
          class="track__element track__element_type_back-end"
        >
          <div
            class="track__bar track__bar_type_back-end"
          >
            1 неделя
          </div>
          <p
            class="track__bar-name"
          >
            Back-end
          </p>
        </div>
        <div
          class="track__element track__element_type_front-end"
        >
          <div
            class="track__bar track__bar_type_front-end"
          >
            4 недели
          </div>
          <p
            class="track__bar-name"
          >
            Front-end
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
